kube-core/src/schema/transform_any_of.rs

```diff
@@ -0,0 +1,238 @@
+
+    use super::*;
+
+    #[test]
+    fn optional_tagged_enum_with_unit_variants() {
+        let original_schema_object_value = serde_json::json!({
+            "anyOf": [
+                {
+                    "description": "A very simple enum with empty variants",
+                    "oneOf": [
+                        {
+                            "type": "string",
+                            "enum": [
+                                "C",
+                                "D"
+                            ]
+                        },
+                        {
+                            "description": "First variant doc-comment",
+                            "type": "string",
+                            "enum": [
+                                "A"
+                            ]
+                        },
+                        {
+                            "description": "Second variant doc-comment",
+                            "type": "string",
+                            "enum": [
+                                "B"
+                            ]
+                        }
+                    ]
+                },
+                {
+                    "enum": [
+                        null
+                    ],
+                    "nullable": true
+                }
+            ]
+        });
+
+        let expected_converted_schema_object_value = serde_json::json!({
+            "description": "A very simple enum with empty variants",
+            "nullable": true,
+            "oneOf": [
+                {
+                    "type": "string",
+                    "enum": [
+                        "C",
+                        "D"
+                    ]
+                },
+                {
+                    "description": "First variant doc-comment",
+                    "type": "string",
+                    "enum": [
+                    "A"
+                ]
+                },
+                {
+                    "description": "Second variant doc-comment",
+                    "type": "string",
+                    "enum": [
+                        "B"
+                    ]
+                }
+            ]
+        });
+
+
+        let original_schema_object: SchemaObject =
+            serde_json::from_value(original_schema_object_value).expect("valid JSON");
+        let expected_converted_schema_object: SchemaObject =
+            serde_json::from_value(expected_converted_schema_object_value).expect("valid JSON");
+
+        let mut actual_converted_schema_object = original_schema_object.clone();
+        hoist_any_of_subschema_with_a_nullable_variant(&mut actual_converted_schema_object);
+
+        assert_json_eq!(actual_converted_schema_object, expected_converted_schema_object);
+    }
+
+    #[test]
+    fn optional_tagged_enum_with_unit_variants_but_also_an_existing_description() {
+        let original_schema_object_value = serde_json::json!({
```

Comment:
i think these tests are nice.

minor nit; variables length here causes my eyes to rescan quite a bit. we can remove the substring `_schema_object` from all the variables here to make it slightly less verbose as all variables have this substring.